task_1.cpp

```diff
@@ -0,0 +1,190 @@
+#include <iostream>
+#include <fstream>
+
+char digit(int num)
+{
+    switch (num)
```

Comment:
I think you can replace this `switch` with 
`if (num < 9) return num + '0'
else return num + 'A' - 10`.
Take a look on [chars table](https://www.cs.cmu.edu/~pattis/15-1XX/common/handouts/ascii.html) and chars indexes